src/gui/fspathedit.cpp

```diff
@@ -111,8 +111,12 @@ FileSystemPathEdit::FileSystemPathEditPrivate::FileSystemPathEditPrivate(
 void FileSystemPathEdit::FileSystemPathEditPrivate::browseActionTriggered()
 {
     Q_Q(FileSystemPathEdit);
+
+    const QString path = QFileInfo(q->selectedPath()).absoluteFilePath();
+    const QString directory = (m_mode == FileSystemPathEdit::Mode::DirectoryOpen) || (m_mode == FileSystemPathEdit::Mode::DirectorySave)
+            ? path
+            : QDir(path).absolutePath();
```

Comment:
It seems you can utilize QFileInfo methods:
```c++
// just an example, I didn't verify correctness
const QFileInfo info(q->selectedPath());
const QString directory = (m_mode == FileSystemPathEdit::Mode::DirectoryOpen) || (m_mode == FileSystemPathEdit::Mode::DirectorySave)
            ? info.absoluteFilePath()
            : info.absolutePath();
```

Reply:
Done